src/math/src/lib.cairo

```diff
@@ -233,3 +235,115 @@ impl U256BitRotate of BitRotate<u256> {
         remainder * pow(2, 256 - n) + quotient
     }
 }
+
+trait WrappingMath<T> {
+    fn wrapping_add(self: T, rhs: T) -> T;
+    fn wrapping_sub(self: T, rhs: T) -> T;
+    fn wrapping_mul(self: T, rhs: T) -> T;
+}
+
+impl U8WrappingMath of WrappingMath<u8> {
```

Comment:
I think it makes more sense to have one trait WrappingAdd, WrappingSub, WrappingMul so that if someone wants to create a function with  a requirement on T implementing any of these operations, it doesn't necessarily need to implement the others.

So, for example (equivalent for checkedmath):

```
    trait CheckedMath<T> {
        fn checked_add(self: T, rhs: T) -> Option<T>;
        fn checked_sub(self: T, rhs: T) -> Option<T>;
    }

    impl CheckedMathImpl<T, +CheckedAdd<T>, +CheckedSub<T>> of CheckedMath<T> {
        fn checked_add(self: T, rhs: T) -> Option<T> {
            CheckedAdd::<T>::checked_add(self, rhs)
        }

        fn checked_sub(self: T, rhs: T) -> Option<T> {
            CheckedSub::<T>::checked_sub(self, rhs)
        }
    }
 ```
 
 and then:
 ```
  trait CheckedAdd<T> {
        fn checked_add(self: T, rhs: T) -> Option<T>;
    }

    impl U8CheckedAdd of CheckedAdd<u8> {
        fn checked_add(self: u8, rhs: u8) -> Option<u8> {
            u8_checked_add(self, rhs)
        }
    }
```

(same for other traits)

Reply:
Great, totally agree. With this, we can both provide a full `WrappingMath` trait for succintness but also traits for the individual ops.

Reply:
Done in [6c6eef1](https://github.com/keep-starknet-strange/alexandria/pull/241/commits/6c6eef1c369c45aa5839c6766dbe0bf7cb123338).